src/Rings/MPolyQuo.jl

```diff
@@ -48,6 +48,10 @@ mutable struct MPolyQuoElem{S} <: RingElem
   P::MPolyQuo{S}
 end
 
+function Base.deepcopy_internal(a::MPolyQuoElem, dict::IdDict)
+  return MPolyQuoElem(Base.deepcopy_internal(a.f, dict), a.P)
+end
```

Comment:
Could also do this, which would remove the need to infer the type parameters for the copy, instead it now force the result to have the right type. This would catch if `Base.deepcopy_internal(a.f, dict)` for some reason returns something of the wrong type (i.e., because of a bug elsewhere).
```suggestion
function Base.deepcopy_internal(a::T, dict::IdDict) where T <: MPolyQuoElem
  return T(Base.deepcopy_internal(a.f, dict), a.P)
end
```

That said, it's hardly important, just wanted to point out the possibility.

Reply:
Noted. Although the most recent 741c11e has
```julia
function Base.deepcopy_internal(a::MPolyElemLoc{T}, dict::IdDict) where T
  return MPolyElemLoc{T}(Base.deepcopy_internal(a.frac, dict), a.parent, false)
end
```